File: example-java/src/main/java/ru/configuration/MyConfig.java
package ru.configuration;

import io.art.core.collection.*;
import io.art.core.source.*;
import lombok.*;
import static io.art.core.constants.CompilerSuppressingWarnings.*;
import java.time.*;
import java.util.*;

@Getter
@ToString
@RequiredArgsConstructor
@SuppressWarnings(ALL)
public class MyConfig {
    private final Integer FBInteger;
    private final Short FBShort;
    private final Byte FBByte;
    private final Boolean FBBoolean;
    private final Long FBLong;
    private final String FBString;
    private final Double FBDouble;
    private final Float FBFloat;
    private final UUID FUUID;
    private final LocalDateTime FlocalDateTime;
    private final ZonedDateTime FzonedDateTime;
    private final Date Fdate;

    private final int[] FAint;
    private final char[] FAchar;
    private final short[] FAshort;
    private final byte[] FAbyte;
    private final boolean[] FAboolean;
    private final long[] FAlong;
    private final double[] FAdouble;
    private final float[] FAfloat;

    private final Integer[] FABInteger;
    private final Character[] FABCharacter;
    private final Short[] FABShort;
    private final Byte[] FABByte;
    private final Boolean[] FABBoolean;
    private final Long[] FABLong;
    private final String[] FABString;
    private final Double[] FABDouble;
    private final Float[] FABFloat;

    private final List<Integer> FLInteger;
    private final List<Character> FLCharacter;
    private final List<Short> FLShort;
    private final List<Byte> FLByte;
    private final List<Boolean> FLBoolean;
    private final List<Long> FLLong;
    private final List<String> FLString;
    private final List<Double> FLDouble;
    private final List<Float> FLFloat;

    private final Set<Integer> FSInteger;
    private final Set<Character> FSCharacter;
    private final Set<Short> FSShort;
    private final Set<Byte> FSByte;
    private final Set<Boolean> FSBoolean;
    private final Set<Long> FSLong;
    private final Set<String> FSString;
    private final Set<Double> FSDouble;
    private final Set<Float> FSFloat;

    private final MyConfig FModel;
    private final List<MyConfig> FModelL;
    private final Set<MyConfig> FModelS;
    private final MyConfig FAModel;

    private final MyConfig[] FModelA;

    private final List<MyConfig[]> FModelLA;
    private final List<MyConfig>[] FModelAL;
    private final List<MyConfig[]>[] FModeALA;

    private final Set<MyConfig[]> FModelSA;
    private final Set<MyConfig>[] FModelAS;
    private final Set<MyConfig[]>[] FModelSAA;

    private final ImmutableArray<MyConfig[]> FModelAA;
    private final ImmutableArray<MyConfig>[] FModelAAA;
    private final ImmutableArray<MyConfig[]>[] FModelAAAA;

    private final ImmutableSet<MyConfig[]> FModelSAAAAA;
    private final ImmutableSet<MyConfig>[] FModelSAAAAAA;
    private final ImmutableSet<MyConfig[]>[] FModelLSAA;

    private final List<ImmutableArray<MyConfig>> FModelLAAAAA;
    private final List<ImmutableArray<MyConfig>[]> FModelLAA;

    private final List<ImmutableSet<MyConfig>> FModelLS;
    private final List<ImmutableSet<MyConfig>[]> FModelLSA;

    private final ImmutableArray<List<MyConfig>> FModelLL;
    private final ImmutableArray<List<MyConfig>[]> FModelLLLA;

    private final ImmutableSet<List<MyConfig>> FModelLSL;
    private final ImmutableSet<List<MyConfig>[]> FModelLSLA;
    private final ImmutableSet<List<MyConfigParent>[]> FPModelLSLA;

    private final ImmutableMap<String, List<MyConfigParent>[]> FPModelLSLAIM;
    private final Map<String, List<MyConfigParent>[]> FPModelLSLAM;
    private final Map<String, Map<String, List<MyConfigParent>>[]> FPModelLSLAMM;
    private final Map<String, Map<String, Optional<List<MyConfigParent>>>[]> FPModelLSLAMOM;

    private final Optional<ImmutableMap<String, Set<String>>> FSIM;
    private final Optional<Map<String, String>> FSM;
    private final Optional<Map<String, String>[]> FSMA;

    private final NestedConfiguration FNested;
    private final Map<String, NestedConfiguration> FNestedMap;
    private final ImmutableMap<String, NestedConfiguration> FNestedIMap;
    private final List<NestedConfiguration> FNestedList;
    private final ImmutableArray<NestedConfiguration> FNestedIList;
    private final Set<NestedConfiguration> FNestedSet;
    private final ImmutableSet<NestedConfiguration> FNestedISet;
}

File: example.kotlin/src/main/java/ru/configuration/MyConfig.kt
package ru.configuration

import io.art.core.collection.ImmutableArray
import io.art.core.source.NestedConfiguration
import java.time.LocalDateTime
import java.time.ZonedDateTime
import java.util.*


data class MyConfig(
        val FBInteger: Int?,
        val FBShort: Short,
        val FBByte: Byte,
        val FBBoolean: Boolean,
        val FBLong: Long,
        val FBString: String,
        val FBDouble: Double,
        val FBFloat: Float,
        val FUUID: UUID,
        val FlocalDateTime: LocalDateTime,
        val FzonedDateTime: ZonedDateTime,
        val Fdate: Date,


        val FAint: IntArray,
        val FAchar: CharArray,
        val FAshort: ShortArray,
        val FAbyte: ByteArray,
        val FAboolean: BooleanArray,
        val FAlong: LongArray,
        val FAdouble: DoubleArray,
        val FAfloat: FloatArray,

        val FABInteger: Array<Int>,
        val FABCharacter: Array<Char>,
        val FABShort: Array<Short>,
        val FABByte: Array<Byte>,
        val FABBoolean: Array<Boolean>,
        val FABLong: Array<Long>,
        val FABString: Array<String>,
        val FABDouble: Array<Double>,
        val FABFloat: Array<Float>,


        val FLInteger: List<Int>,
        val FLCharacter: List<Char>,
        val FLShort: List<Short>,
        val FLByte: List<Byte>,
        val FLBoolean: List<Boolean>,
        val FLLong: List<Long>,
        val FLString: List<String>,
        val FLDouble: List<Double>,
        val FLFloat: List<Float>,


        val FSInteger: Set<Int>,
        val FSCharacter: Set<Char>,
        val FSShort: Set<Short>,
        val FSByte: Set<Byte>,
        val FSBoolean: Set<Boolean>,
        val FSLong: Set<Long>,
        val FSString: Set<String>,
        val FSDouble: Set<Double>,
        val FSFloat: Set<Float>,
        val FModelL: List<MyConfig>,
        val FModelS: Set<MyConfig>,
        val FModelA: Array<MyConfig>,


        val FModelLA: List<Array<MyConfig>>,
        val FModelAL: Array<List<MyConfig>>,
        val FModeALA: kotlin.Array<List<Array<MyConfig>>>,


        val FModelSA: Set<Array<MyConfig>>,
        val FModelAS: kotlin.Array<Set<MyConfig>>,
        val FModelSAA: kotlin.Array<Set<Array<MyConfig>>>,


        val FModelAA: ImmutableArray<Array<MyConfig>>,
        val FModelAAA: kotlin.Array<ImmutableArray<ru.configuration.MyConfig>>,
        val FModelAAAA: kotlin.Array<ImmutableArray<Array<MyConfig>>>,


        val FModelSAAAAA: io.art.core.collection.ImmutableSet<Array<MyConfig>>,
        val FModelSAAAAAA: kotlin.Array<io.art.core.collection.ImmutableSet<ru.configuration.MyConfig>>,
        val FModelLSAA: kotlin.Array<io.art.core.collection.ImmutableSet<Array<MyConfig>>>,

        val FModelLAAAAA: List<ImmutableArray<ru.configuration.MyConfig>>,
        val FModelLAA: List<kotlin.Array<ImmutableArray<ru.configuration.MyConfig>>>,
        val FModelLS: List<io.art.core.collection.ImmutableSet<ru.configuration.MyConfig>>,
        val FModelLSA: List<kotlin.Array<io.art.core.collection.ImmutableSet<ru.configuration.MyConfig>>>,
        val FModelLL: ImmutableArray<List<MyConfig>>,
        val FModelLLLA: ImmutableArray<Array<List<MyConfig>>>,


        val FModelLSL: io.art.core.collection.ImmutableSet<List<MyConfig>>,
        val FModelLSLA: io.art.core.collection.ImmutableSet<Array<List<MyConfig>>>,
        val FPModelLSLA: io.art.core.collection.ImmutableSet<kotlin.Array<List<ru.configuration.MyConfigParent>>>,


        val FPModelLSLAIM: io.art.core.collection.ImmutableMap<kotlin.String, kotlin.Array<List<ru.configuration.MyConfigParent>>>,
        val FPModelLSLAM: Map<kotlin.String, kotlin.Array<List<ru.configuration.MyConfigParent>>>,
        val FPModelLSLAMM: Map<String, Array<Map<String, List<MyConfigParent>>>>,
        val FPModelLSLAMOM: Map<String, Array<Map<String, Optional<List<MyConfigParent>>>>>,


        val FSIM: java.util.Optional<io.art.core.collection.ImmutableMap<kotlin.String, Set<kotlin.String>>>,
        val FSM: java.util.Optional<Map<String, String>>,
        val FSMA: java.util.Optional<kotlin.Array<Map<String, String>>>,


        val FNested: NestedConfiguration,
        val FNestedMap: Map<String, NestedConfiguration>,
        val FNestedIMap: io.art.core.collection.ImmutableMap<kotlin.String, NestedConfiguration>,
        val FNestedList: List<NestedConfiguration>,
        val FNestedIList: ImmutableArray<NestedConfiguration>,
        val FNestedSet: Set<NestedConfiguration>,
        val FNestedISet: io.art.core.collection.ImmutableSet<NestedConfiguration>?,
)
